runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -221,29 +265,85 @@ public void updateStreamTails(long address, ILogData logData) {
         }
     }
 
+    /**
+     * It is a valid state to have entries that are checkpointed but the log was not fully
+     * trimmed yet. This means that some entries are at the same time in their own slot and in
+     * the checkpoint. We must avoid to process them twice.
+     *
+     * This comes especially relevant when the operation order affects the end result.
+     * (e.g. clear() operation)
+     *
+     * @param streamId stream id to validate
+     * @param entry entry under scrutinization.
+     * @return if the entry is already part of the checkpoint we started from.
+     */
+    private boolean entryAlreadyContainedInCheckpoint(UUID streamId, SMREntry entry) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "SMREntry"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)